### ISSUE TITLE
Modifying nested data

### ISSUE THREAD

**Issue by davidkhess** [vuejs/Discussion#282]
When modifying nested data, you don't always get the expected results. See this fiddle:

http://jsfiddle.net/davidkhess/fwr8756v/2/

Using the straightforward and documented $set on the nested data doesn't get the expected results as the observers at the top level don't get notified of the change.

What's the right answer here? When nested data is modified, make sure and set the value at the top level again?

This is a subtle issue that should probably be documented along with $set since $set doesn't appear to work in this case. The $set docs do mention using paths which sort of alludes to it but doesn't address it as a requirement for nested data.

**Comment by davidkhess** [comment]
Note, this is not a problem with nested objects:

http://jsfiddle.net/davidkhess/fwr8756v/3/

Is it possible this is a bug where arrays are not properly observing their contents?

**Comment by yyx990803** [comment]
I can explain some details. In particular, an object has to invoke its own associated dependency when a new property is added, but when its accessed as an element in an Array, that dependency is not collected.

``` js
a.b // b's dep is collected because a.b invokes the getter
arr[0] // arr[0]'s dep not collected because Array index access cannot be intercepted
```

**Comment by davidkhess** [comment]
Would a $get on Array help the situation? But would it cover the case of a v-repeat over an array of objects if a repeated child component modified one of the objects?

Regardless, is there a good pattern for mitigating this without any changes to Vue? Should you emit an event to the parent view model to force an update at the top level?

**Comment by yyx990803** [comment]
Fixed in `dev`: 171fdc7ad5a613331155a865056d2fbc591e4693

Now it should just work.

**Comment by davidkhess** [comment]
Wow, great. Thanks!
